src/Cli/dotnet/CommonArguments.cs

```diff
@@ -0,0 +1,51 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+
+using System.CommandLine;
+using System.CommandLine.Parsing;
+using Microsoft.DotNet.Cli.Utils;
+using NuGet.Packaging.Core;
+using NuGet.Versioning;
+
+namespace Microsoft.DotNet.Cli
+{
+    internal class CommonArguments
+    {
+        #region PackageIdentityArgument
+        public static CliArgument<PackageIdentity> PackageIdentityArgument(bool requireArgument = true) =>
+            new("packageId")
+            {
+                HelpName = "PACKAGE_ID",
+                Description = CliStrings.PackageIdentityArgumentDescription,
+                CustomParser = ParsePackageIdentity,
+                Arity = requireArgument ? ArgumentArity.ExactlyOne : ArgumentArity.ZeroOrOne,
+            };
+
+        private static PackageIdentity ParsePackageIdentity(ArgumentResult argumentResult)
+        {
+            if (argumentResult.Tokens.Count == 0)
```

Comment:
PackageIdentityArgument doesn't require an argument.

If it gets here, that means it has no argument, which is explicitly permitted, and returns null.

But then we have checks (like in ToolUpdateLocalCommand) that do parseResult.GetValue(stuff).Id.

That looks like a NRE to me